Project/Source/StateMachineManager.cpp

```diff
@@ -170,26 +189,32 @@ bool StateMachineManager::CalculateAnimation(GameObject* gameObject, const GameO
 		}
 
 		//Sending event on keyframe
-		if (!clip->keyEventClips.empty()) { //Only call this once
+		unsigned int currentEventKeyFrame = componentAnimation.listClipsCurrentEventKeyFrames[currentState.clipUid];
+		if (!componentAnimation.listClipsKeyEvents[currentState.clipUid].empty()) { //Only call this once
 			// Send key Frame event
-			for (int difference = currentSample - clip->currentEventKeyFrame; difference <= currentSample; difference++) {
-				if (clip->keyEventClips.find(difference) != clip->keyEventClips.end() && !clip->keyEventClips[difference].sent) {
+			int difference = currentSample - currentEventKeyFrame;
+			int i = 0;
+			if (currentEventKeyFrame == 178) {
```

Comment:
What is this for? I don't see the int a used anywhere

Reply:
Removed